types/yup/yup-tests.ts

```diff
@@ -432,9 +432,56 @@ const description: SchemaDescription = {
     label: 'label',
     meta: { key: 'value' },
     tests: [{ name: 'test1', params: {} }, { name: 'test2', params: {} }],
-    fields: { key: 'value' },
+    fields: { key: { type: 'ref', key: 'value' } },
 };
 
+const description2: SchemaDescription = {
+    type: 'type',
+    label: 'label',
+    meta: { key: 'value' },
+    tests: [{ name: 'test1', params: {} }, { name: 'test2', params: {} }],
+    fields: {
+        another: {
+            type: 'type',
+            label: 'label',
+            meta: { key: 'value' },
+            tests: [],
+        },
+    },
+};
+
+const description3: SchemaDescription = {
+    type: 'type',
+    label: 'label',
+    meta: { key: 'value' },
+    tests: [{ name: 'test1', params: {} }, { name: 'test2', params: {} }],
+    fields: {
+        another: {
+            type: 'type',
+            label: 'label',
+            meta: { key: 'value' },
+            tests: [],
+            fields: { key: { type: 'ref', key: 'value' } }
+        },
+    },
+};
+
+ const description4: SchemaDescription = {
```

Comment:
It's a bit confusing what's being asserted in `description2`, `description3`, and description4` as the objects are 90% the same. Would it make sense to do all these variations as different fields on the original `description` variable?

Reply:
Good idea, fixed.